jake/__main__.py

```diff
@@ -334,37 +334,47 @@ def __iq_control_flow(args: dict, bom_str: bytes):
     spinner.text = "Reticulating splines..."
     iq_requests.poll_report(status_url)
 
-    if iq_requests.get_policy_action() == 'Failure':
-      spinner.fail("💥 ")
-      __toggle_stdout(on=True)
-      print(Fore.YELLOW +
-            "Snakes on the plane! There are policy failures from Sonatype IQ.")
-      print(Fore.YELLOW +
-            "Your IQ Server Report is available here: {}".format(iq_requests.get_report_url()))
-      _exit(1)
-    elif iq_requests.get_policy_action() == 'Warning':
-      spinner.ok("🧨 ")
-      __toggle_stdout(on=True)
-      print(Fore.GREEN +
-            "Something slithers around your ankle! There are policy warnings from Sonatype IQ.")
-      print(Fore.GREEN +
-            "Your IQ Server Report is available here: {}".format(iq_requests.get_report_url()))
-      _exit(0)
-    elif iq_requests.get_policy_action() == 'None':
-      spinner.ok("🐍 ")
-      __toggle_stdout(on=True)
-      print(Fore.GREEN +
-            "Smooth slithering there bud! No policy failures from Sonatype IQ.")
-      print(Fore.GREEN +
-            "Your IQ Server Report is available here: {}".format(iq_requests.get_report_url()))
-      _exit(0)
-    else:
-      spinner.fail("🐍 ")
-      __toggle_stdout(on=True)
-      print(Fore.RED +
-            "Received an error response from IQ Server, please check logs. policy action: {}"
-            .format(iq_requests.get_policy_action()))
-      _exit(3)
+    exit_status_code = _show_summary(iq_requests, spinner)
+    _exit(exit_status_code)
```

Comment:
YOU COULD inline the call inside this _exit, but it's no big deal otherwise. I think it reads cleaner that way you did it, even if it's just a throwaway variable. More explicit what comes back from the call, etc...

Reply:
Good point! I went ahead a removed the variable. 0c5a609